openid-4-verifiable-presentations-1_0.md

```diff
@@ -1464,6 +1464,132 @@ issuers in Self-Sovereign Identity ecosystems using TRAIN</title>
+
+## Protocol
+
+The value of the `protocol` parameter of the W3C Digital Credentials API MUST be set to `urn:openid.net:oid4vp' for this profile.
+
+## Request
+
+The `request` parameter of the W3C Digital Credentials API MUST contain a valid OID4VP Authorization Request, where every Authorization Request parameters is represented as top-level JSON claim as shown in the following non-normative example:
+
+```json
+{
+  "client_id": "client.example.org",
+  ...
+  "response_type": "vp_token",
+  "nonce": "n-0S6_WzA2Mj",
+  "presentation_definition": {...}
+}
+```
+
+Any Authorization Requests as defined in OID4VP can be used except `redirect_uri`. If present, `redirect_uri` MUST be ignored. 
+
+NOTE: Implementations MAY use the `request_uri` request parameter as well as the response mode `direct_post` if deeemd necessary. If the `request_uri` parameter is used, the `request` value MUST nevertheless contain the `presentation_definition` parameter in order to enable wallet selection. The signed request object, fetched from the `request_uri` MUST contain a `presentation_definition` that is compatible with the `presentation_definition` in the `request` parameter of the W3C Digital Credentials API. 
+
+### Client ID Scheme web-origin
+
+This profile defines the new Client ID Scheme `web-origin`. If this scheme is used, the Client Identifier is the origin of the Verifier web site as asserted by the browser towards the Wallet. 
+
+## Response
+
+Every OID4VP request MUST result in a response being provided through the W3C Digital Credentials API. In all cases, except if the response mode `direct_post` is used, the response MUST be follow the rules specified for OID4VP Authorization Responses. If the response model `direct_post` is used, the Wallet will return the URL the Direct Post Endpoint of the Verifier has returned. 
+
+The following is an example of an OID4VP Authorization Response through the API: 
+
+```json
+{
+  "presentation_submission": "...",
```

Comment:
Note that, if you respond with a JSON payload, then the new SearchURLParams() decoding isn't what you want, but rather JSON.parse().

Reply:
As I understand OID4VP and because it is "a mechanism on top of OAuth 2.0" the response is something resulting in an HTTP request to the `redirect_uri`. Use in the browser API is different, obviously, and a JSON payload likely makes sense. But I don't think that's congruent with "response MUST be follow the rules specified for OID4VP Authorization Responses". I believe a lot more is needed in this section to reconcile how responses/response_mode should be conveyed in this context and how it's different than defined in typical OID4VP/OAuth.  Maybe new response modes or describing a mapping of how protocol parameters are transmitted. Sorry, this is difficult to articulate but the defined authorization response mechanisms don't just work as-is in a browser API context and Sam's comment here kinda highlights that or is a symptom of it.

Reply:
clarified that the response is a JSON object and the response type specific response parameters are encoded as JSON claims of that object

Reply:
should something also be said about response mode? JARM's `jwt` would/could indicate signed and/or encrypted that would look like: `{"response":"eyJraWQiOiJsY...hLWfug6dA"}`.  While the query, fragment, and form_post vairients (JARM and plain) are kinda no ops or maybe disallowed. Direct post doesn't make sense (does it?) so maybe disallowed. Or is/should this be new response mode(s)? like `api` and `api.jwt` or something like that. It almost just makes sense but I think something needs to be said or defined in this area for interoperability.

Reply:
WG discussion.
two questions that are related but should probably be dealt separately:
1. when response is encrypted, whether it is JARM extension in OpenID4VP (that allows only encrypting the response) or not? agreed to make changes to the PR with this as a starting point. 
2. what is the response_mode that should be used with the browser API? do not include response_mode at all (that would make it very implicit whether encryption is required or not)? agreed to make changes to the PR with new response_mode parameters `w3c_dc_api` and `w3c_dc_api.jwt`. an option is to say that if it is omitted, the deafult is `api.

Reply:
these changes have been applied to the draft